salt/metalk8s/kubernetes/etcd/installed.sls

```diff
@@ -4,19 +4,33 @@ include:
   - metalk8s.kubernetes.ca.etcd.advertised
   - .certs
 
-{%- set host_name = grains['id'] %}
-{%- set host = grains['metalk8s']['control_plane_ip'] %}
+{%- set node_name = grains['id'] %}
+{%- set node_ip = grains['metalk8s']['control_plane_ip'] %}
 
-{%- set endpoint  = host_name ~ '=https://' ~ host ~ ':2380' %}
+{%- set endpoint = 'https://' ~ node_ip ~ ':2380' %}
 
-{#- Get the list of existing etcd node. #}
-{%- set etcd_endpoints = pillar.metalk8s.etcd.members %}
+{#- Get the list of existing etcd member. #}
+{%- set etcd_members = pillar.metalk8s.etcd.members %}
 
 {#- Compute the initial state according to the existing list of node. #}
-{%- set state = "existing" if etcd_endpoints else "new" %}
+{%- set state = "existing" if etcd_members else "new" %}
 
-{#- Add ourselves to the list. #}
-{%- do etcd_endpoints.append(endpoint) %}
+{%- set etcd_endpoints = {} %}
+{%- for member in etcd_members %}
+  {#- NOTE: Filter out members with empty name as they are not started yet. #}
+  {%- if member['name'] %}
+    {#- NOTE: Only take first peer_urls for endpoint. #}
+    {%- do etcd_endpoints.update({member['name']: member['peer_urls'][0]}) %}
+  {%- endif %}
+{%- endfor %}
+
+{#- Add ourselves to the endpoints. #}
+{%- do etcd_endpoints.update({node_name: endpoint}) %}
+
+{%- set etcd_initial_cluster = [] %}
+{%- for name, ep in etcd_endpoints.items() %}
```

Comment:
I think it's better to not break the etcd cluster if we have one node that changed peer urls (since we only support one peer urls here)

Unfortunatly I didn't found any good method to do it without this second loop, if you have an idea do not hesitate
At the beginning just wanted to do something like that in python directly for `initial-cluster`
```
','.join('{}={}'.format(name, ip) for name, ip in etcd_endpoints.items())
```
But not doable in jinja without a custom filter

Reply:
My point was that we're supposed to have a unique set of members (with unique peer_urls, I'd imagine), so we could have just done:

```jinja
{%- set initial_cluster = [] %}
{%- for member in etcd_members | selectattr('name') %}
  {%- do initial_cluster.append(member['name'] ~ '=' ~ member['peer_urls'][0]) %}
{%- endfor %}

{%- do initial_cluster.append(node_name ~ '=' ~ endpoint %}
```

But it's really minor, and your approach is safer against duplicate values in `etcd_members`, so let's leave it as is.